src/Insight.Box.FrontEnd/src/modules/accounts/companents/AuthModal.vue

```diff
@@ -0,0 +1,133 @@
+                                :placeholder="LayoutConstants.EnterPassword" :type="FormInputType.Password"/>
+
+                    <!-- Modal actions -->
+                    <div class="flex gap-10">
+                        <app-button :type="ButtonType.Secondary" class="w-full" :text="LayoutConstants.Cancel"
+                                    @click="closeModal"/>
+                        <app-button :type="ButtonType.Primary" class="w-full" :text="LayoutConstants.Submit"
+                                    :role="ButtonRole.Submit" @click="onSubmit"/>
+                    </div>
+
+                </form>
+            </div>
+
+        </template>
+
+    </ModalBase>
+
+</template>
+
+<script setup lang="ts">
+
+import {defineEmits, ref, watch} from 'vue';
+import {LayoutConstants} from "@/common/constants/LayoutConstants";
+import FormInput from "@/common/components/formInput/FormInput.vue";
+import ModalBase from "@/common/components/modalBase/ModalBase.vue";
+import {SignUpDetails} from "@/modules/accounts/models/SignUpDetails";
+import {SignInDetails} from "@/modules/accounts/models/SignInDetails";
+import {FormInputType} from "@/common/components/formInput/FormInputType";
+import {InsightBoxApiClient} from "@/infrastructure/apiClients/insightBoxClient/brokers/InsightBoxApiClient";
+import {AuthenticationService} from "@/modules/accounts/services/AuthenticationService";
+import {ButtonType} from "@/common/components/appButton/ButtonType";
+import {ButtonRole} from "@/common/components/appButton/ButtonRole";
+import AppButton from "@/common/components/appButton/AppButton.vue";
+
+const  authService = new AuthenticationService();
+const insightBoxApiClient = new InsightBoxApiClient();
+
+const props = defineProps({
+    isActive: {
+        type: Boolean,
+        default: false
+    }
+});
+
+
+const emit = defineEmits<{
+    (e: 'submit', signInDetails: SignInDetails): void
+    (e: 'closeModal'): void
+}>();
+
+const signUpDetails = ref<SignUpDetails>(new SignUpDetails());
+const signInDetails = ref<SignInDetails>(new SignInDetails());
+const modalTitle = ref<string>(LayoutConstants.WelcomeToMessage);
+const isSignIn = ref<boolean>(true);
+const foundUserName = ref<string>('');
+
+const closeModal = () => {
+    // Reset values;
+    if(props.isActive == true)
+         emit('closeModal');
+}
+
+const checkIfEmailAddressExists = async (emailAddress: string) => {
+    foundUserName.value = '';
+    const response = await insightBoxApiClient.clients.checkByEmailAsync(emailAddress);
+    if(response.response)
+        foundUserName.value = response.response;
+
+    return foundUserName.value;
+}
+
+const onSubmit = async () => {
+    // If user tried to sign in, but email address not found
+    if(isSignIn.value && !foundUserName.value && !(await checkIfEmailAddressExists(signInDetails.value.emailAddress))) {
+        isSignIn.value = false;
+        modalTitle.value = LayoutConstants.WelcomeToMessage;
+        return;
+    }
```

Comment:
What if we are in sign in mode and user name is found ? We have 2 issues with this block :

- it doesn't set username in that case
- it passes to the sign in method and causes 500 because user password is not entered yet